cmake/finds/RequiredDependencySearch.cmake

```diff
@@ -74,11 +74,15 @@ GetDependency(OTF2)
 
 ########################################
 ##### LOOK FOR AND/OR BUILD HDF5  ######
-########################################
 
 set(HAVE_HDF5 1)
 
-GetDependency(HDF5)
+find_package(HDF5 REQUIRED COMPONENTS C CXX HL)
+if (HDF5_FOUND)
```

Comment:
since there is the `REQUIRED` parameter in the `find_package` call, [it will error out if not found](https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature).

Reply:
Ok so why the need for IF_found?